src/audio/component.c

```diff
@@ -119,8 +119,15 @@ void comp_unregister(struct comp_driver *drv)
 
 int comp_set_state(struct comp_dev *dev, int cmd)
 {
+	int requested_state = comp_get_requested_state(cmd);
 	int ret = 0;
 
+	if (dev->state == requested_state) {
+		trace_comp("comp_set_state(), state already set to %u",
+			   dev->state);
+		return 1;
+	}
```

Comment:
I wonder if it wouldn't be better to just return 0 here. There is no distinction in the code between "state was changed" and "state is already as it should be".

With this, we only have two possible values for comp_set_state returned value: 0 for success, negative value for error.

Reply:
It returns 1 on purpose to stop walking pipeline graph any further. It is assumed, that current pipeline has been already set to the desired state. Desired in scenario with multiple connected pipelines. Probably such handling should be revisited and refactored in the future.

Reply:
@tlauda makes sense. Thanks for the explanation.